### ISSUE TITLE
False-positive for IDE0251 Make member readonly

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn#67531]
**Version Used**: '4.6.0-2.23151.17 (1314d090)'. Language version: 11.0.

**Steps to Reproduce**:

````c#
struct Repro
{
    private unsafe fixed byte bytes[16];

    // IDE0251 suggests making this method readonly
    // If you add readonly, the compiler reports
    // CS1605 Cannot use 'bytes[0]' as a ref or out value because it is read-only
    public unsafe Span<byte> AsSpan()
    {
        return MemoryMarshal.CreateSpan(ref bytes[0], 16);
    }
}
````

**Diagnostic Id**: IDE0251 Make member 'readonly'

**Expected Behavior**: The diagnostic is not reported

**Actual Behavior**: The diagnostic is reported

**Comment by CyrusNajmabadi** [comment]
Thanks for the repro!

**Comment by jeremy-visionaid** [comment]
Is this fixed being shipped? I'm still seeing it, or something very similar in VS 17.6.5 with record structs:

```
    public record struct DownloadProgressInfo(long BytesDownloaded, long TotalLength)
    {
        public int PercentComplete => (int)Math.Round(100.0 * BytesDownloaded / TotalLength);
    }
```

**Comment by CyrusNajmabadi** [comment]
@jeremy-visionaid can you clarify what issue it is you are seeing?

**Comment by jeremy-visionaid** [comment]
@CyrusNajmabadi Don't worry, my bad - ignore me. The property should be readonly for a record struct, the analyzer is correct.

```
   public record struct DownloadProgressInfo(long BytesDownloaded, long TotalLength)
    {
        public readonly int PercentComplete => (int)Math.Round(100.0 * BytesDownloaded / TotalLength);
    }
```

**Comment by CyrusNajmabadi** [comment]
Ok thanks!
